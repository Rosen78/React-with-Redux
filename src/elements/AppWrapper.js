/**
 * Created by Rosen_Nikolov on 12/23/2016.
 */
import React, { Component} from 'react';
import {browserHistory} from 'react-router';
import {START_URL_PATH} from '../constants/additionalConstants';

export default class AppWrapper extends Component {

    constructor(props) {
        super(props);
    }

    componentWillMount() {
        let path = browserHistory.getCurrentLocation().pathname;

        if(path === '/') {
            browserHistory.push(encodeURI(START_URL_PATH));
        }
    }

    render() {
        return(
            <div>{this.props.children}</div>
        )
    }
}

