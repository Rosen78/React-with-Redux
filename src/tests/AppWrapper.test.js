/**
 * Created by Rosen_Nikolov on 1/9/2017.
 */

import React from 'react';
import renderer from 'react-test-renderer'
import AppWrapper from './../elements/AppWrapper'

it('should be rendered correctly', () => {
    const component = renderer.create(
        <AppWrapper />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});