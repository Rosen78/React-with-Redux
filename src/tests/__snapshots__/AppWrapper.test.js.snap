exports[`test should be rendered correctly 1`] = `<div />`;
